### DISCUSSION TITLE
Another implementation of asChild

### DISCUSSION THREAD

**Discussion by teleskop150750** [unovue/reka-ui#1324] [Ideas]
Implementing asChild via slot seems like a more native and logical option in VueJs philosophy. The slot in Vue combines props.children() and props, which return rendering functions in frameworks like React and Solid.

Positive

- No need for Slot Component anymore.
- [Vapor friendly](https://vapor-repl.netlify.app/#__PROD____VAPOR__eNqFVE1v2zAM/SuEekgCdPZhO2VJgC3ooQP2gW3YSRfXURO1siRItJchyH8f9WHPXtP2EEAiHx+f6Mec2Adri64VbMlWvnbSIniBrd1wLRtrHMLWNBbunWmAs6IMt4Dn7D3XqzKVEJguKBqrKhR0A1jtZBcPdAw1+Qxwq7VwORHZLqIGMriq/PYg1W7Nma+NpcYDaNIlB+5aRKOhe3Mn9VBTVIjOU+XnSpLohBnTlGOeVTl5yURnjxxh2DVD6qLv5b548EbTJE8ByllNVVIJ99WiNJr6LyFmQq5Syvz+FGPoWnHdx+uDqB8vxB/8McQ4++aEF66jOQw5rNxeYErf/PgijnQeko3ZtYrQLyS/C29UGzQm2MdW70j2CBfV3kY/SL3/6W+OKLTvHxWEBuQ54jkje4SBPff0f3LfFu9iHddnmmJvrQtWpPF6hPgVYZ2YjN4qWT8uYS46oXEB603fIqCNEoUy+5QsUssFWTapPL9iXq8Mgq4aQQ7K/uMMllFACGU7ZXv877mcBvxjA0FKEzzRlpuJA0kGRZ9YipjEsThgo0bjyIT9Jx05OO/qCWoniOFXRVfabDinzZ3RWMsuBGdPagIsgYoy/xmMQFO+Of0WRWNajfPZVWXtbDGMMewiyPh+a8Nr46qw819j1GtP) since there is no cloneVNode

Minuses

- Some cost for creating slot
- No root node. Default attribute inheritance no longer works, getting templateRef requires additional work
- Duplicate attributes

[Reference](https://ark-ui.com/react/docs/guides/composition) Solid demo

[Playground](https://play.vuejs.org/#__PROD__eNqFUk1PIzEM/StW9rJI7MxhOZW2ElQcQOJDwDGXYWpKIJNEiWcoQv3vOMnM0KICN8fv2X523rs4ca7oWhQTMQ21V44gILVuLo1qnPUEC9s4ePS2ASmKMr4iX4pjaaZlLmEyPwgbpytCfgFMl6pLAYexpo8Bzo1B3wOp217W2Az+VGHxpPRyJkWorePBI2lnSp94aImsge7fgzJjTVER+cCVl5Vi0Zmz3abc7jMtdzbZ0TkwtzjiUBBPMY9qVTwHa/iS75EqRc1VSqO/dqSs4fkTSEjEKq3t60XKkW/xcMjXT1i/7Mk/h3XMSXHjMaDv+A4jRpVfIWX47O4K1xyPYGOXrWb2D+AtBqvbqDHTTluzZNlbvKT2PPlBmdV9OFsTmjAsFYVG5ibxpWB7xIN9t/qn3P/FUaqTZsNXHKy1x4p83kCQfhFmuZM1C63qlwn8xQ4NHcBsPoyIbKux0HaVwSKPPGDLZpWbX8wbtCUwVYPsoN5/UsAkCYip3k69Pb56roeB3lxskGGm57blfMeBLIOzXyy1+QDotSoE)

**Comment by zernonia** [comment]
I think the usage for `Slot` is still needed, and can't be replaced by Vue native slot. I've written some difference [here](https://www.radix-vue.com/utilities/slot.html) 😁

> [Reference](https://ark-ui.com/react/docs/guides/composition) Solid demo

This link seems to point to Ark UI's composition, which is similar to what we currently have, perhaps you paste the wrong link 🤔

**Reply by teleskop150750** [reply]
> I think the usage for `Slot` is still needed, and can't be replaced by Vue native slot. I've written some difference [here](https://www.radix-vue.com/utilities/slot.html) 😁
> 
> > [Reference](https://ark-ui.com/react/docs/guides/composition) Solid demo
> 
> This link seems to point to Ark UI's composition, which is similar to what we currently have, perhaps you paste the wrong link 🤔

for Vue, slot.asChild is an anlog of props.asChild in Solid.

therefore, I gave the example of Solid, in which mergeProps occurs manually and is controlled by the user

```tsx
import { Button } from '@acme/ui-lib'
import { Popover } from '@ark-ui/solid/popover'

export const Basic = () => (
  <Popover.Root>
    <Popover.Trigger asChild={(props) => <Button {...props()} />}>Open</Popover.Trigger>
  </Popover.Root>
)
```
